### ISSUE TITLE
indexing using a numpy in array?

### ISSUE THREAD

**Issue by funnydevnull** [serge-sans-paille/pythran#677]
Hi, I'm getting an error trying to compile the following code:
```
#pythran export make_vecs(int32[][][], float[][], int list list list, int)
def make_vecs(indvecs, M, ft, nf):
    for si,sf in enumerate(ft):
        for i in range(nf):
            if len(sf[i]) == 0:
                continue
            indvecs[si, i, :len(sf[i])] += sf[i]
    return M[indvecs]
```
The error is long but the relevant last section is:
```
/usr/local/lib/python2.7/dist-packages/pythran/pythonic/types/numpy_texpr.hpp:251:41: note:   no known conversion for argument 1 from 'pythonic::types::numpy_iexpr<const pythonic::types::ndarray<int, 3ul>&>' to 'pythonic::types::ndarray<int, 3ul>::dtype {aka int}'
```

I presume this is related to indexing using a numpy int array.  If I remove the last line above (the return) the code compiles fine and it works (in the sense that it correctly modifies indvecs).

Is this a known limitation or a bug?  Thanks for any help or advice.

**Comment by serge-sans-paille** [comment]
Even 

```
#pythran export make_vecs(int32[][], float[][])
def make_vecs(indvecs, M):
    return M[indvecs]
```

reproduces the issue, I'll check this.

**Comment by serge-sans-paille** [comment]
I added more test cases in #679 but I'm basically not going to support this in a near future; Advanced indexing is a real pain :-/

I added a static_assert that should make the error message slightly more helpful though.

**Comment by funnydevnull** [comment]
Hi Serge,
Thanks for the quick response.  I've been trying to implement something similar without (or with less) fancy indexing but I'm still running into problems.

The following code:
```
#pythran export make_vecs2(float32[][][], float[][], int list list list, int)
def make_vecs2(fvecs, M, ft, nf):
    fvecs *=0
    # iterate over each sentence in the our feature data
    for si,sf in enumerate(ft):
        # add the indices of the features to the index tensor
        for i in range(nf):
            if len(sf[i]) == 0:
                continue
            inds=np.array(sf[i])
# does not work
            fvecs[0, 0] += M[inds].sum(axis=0)
# does not work
#            fvecs[si, i] += M[inds].sum(axis=0)
# works
            #x = M[sf[i]].sum(axis=0)
```

Gives an error:

`PythranTypeError: Invalid subscripting of `int` by `Tuple[int, T12, T13]``

The problem is definitely related to trying to sub-index a 3-index float32 tensor using only two indices.  So its the fvecs[int,int] that's somhow causing a problem.  Is this expected behaviour.

The code works fine in python/numpy.

**Comment by serge-sans-paille** [comment]
mmmh, I think I can warkaround this one, let me have a look!

**Comment by serge-sans-paille** [comment]
Does #680 fixes your issue?

**Comment by funnydevnull** [comment]
I'm not sure what #680 is about.  The fix turns out to be very simple.  The issue is that fvec *= 0 incorrectly converts fvec into an int!  If I replace that line with fvec = fvec * 0 it works.

For reference the pythranized version of this code is ~3x faster than native python even though most of the time is spent in the numpy .sum.

**Comment by serge-sans-paille** [comment]
On Fri, May 19, 2017 at 11:58:43AM -0700, funnydevnull wrote:
> I'm not sure what #680 is about.  The fix turns out to be very simple.  The issue is that fvec *= 0 incorrectly converts fvec into an int!  If I replace that line with fvec = fvec * 0 it works.

Yeah, that's what the fix does : correct type inference for augmented
assign :-)

Glad the performance are there :-)²:x

**Comment by serge-sans-paille** [comment]
Closed by #680
